boilerplate/app/i18n/i18n.ts

```diff
@@ -13,56 +13,51 @@ import fr from "./fr"
 import ja from "./ja"
 import hi from "./hi"
 
-// to use regional locales use { "en-US": enUS } etc
 const fallbackLocale = "en-US"
 
-export let i18n: i18next.i18n
+const systemLocales = Localization.getLocales()
 
-const systemLocale = Localization.getLocales()[0]
-const systemLocaleTag = systemLocale?.languageTag ?? fallbackLocale
+const resources = { ar, en, ko, es, fr, ja, hi }
+const supportedTags = Object.keys(resources)
+
+// Checks to see if the device locale matches any of the supported locales
+// Device locale may be more specific and still match (e.g., en-US matches en)
+const systemTagMatchesSupportedTags = (deviceTag: string) => {
+  const primaryTag = deviceTag.split("-")[0]
+  return supportedTags.includes(primaryTag)
+}
```

Comment:
What it be worth writing a test for this one?

Reply:
I was thinking splitting on `-` and checking `includes` were simple enough it didn't need a test, if we did something more complex maybe then?

Most of the complexity here comes in with third party systems, so an E2E test to check that we can switch languages makes the most sense to me.

Reply:
Yeah, I agree this is not complex, but I'm also thinking of future Felipe doing an i18 migration again. It's all good; not a blocker.

Reply:
For sure. I still think in that case what is most useful is an E2E test instead of unit tests.

Reply:
Filed issue for this: https://github.com/infinitered/ignite/issues/2789